### DISCUSSION TITLE
Increasing open participation

### DISCUSSION THREAD

**Discussion by patricklodder** [dogecoin/dogecoin#2861] [General]
### Ancient history

After I coded my first few pull requests for the Dogecoin software (before it was called Dogecoin Core) in 2014, Ross asked me if I was interested to continue contributing to it and help maintaining the software. The maintainers back then could use the help because the development of the coin was just [handed over to a broader development community](https://web.archive.org/web/20140411193346/http://www.reddit.com/r/dogecoin/comments/22hqm6/official_an_update_on_the_foundation_core) by Jackson Palmer and a lot of work needed to be done to clean the software up and make it maintainable. I accepted after a very quick deliberation because I thought that the upsides - overall nice community, no pre-mine, no ICO, no central governance and a potentially interesting experiment - were more than making up for the downsides: “coinomics” that were at the time (pre-AuxPoW) very problematic, and the community was extremely sensitive to cults of persona and with that vulnerable to scammers. 

This process, where trust was earned based on contributions to the code, is in my opinion essential to have a healthy Dogecoin Core. Dogecoin thrives because of its diverse community, but this is not reflected in the development of it.

### 2020-2021

Between then and 2020, a lot had changed. I personally was inactive as a developer for 3 years and when I came back - because the software was broken and I and others could not easily develop on Dogecoin anymore - I found a new reality, mostly one of complacency. At Sporklin’s insistence, instead of just doing some work and opening pull requests, I agreed to joining the Discord group where devs & co were hanging out and as I took in what was going on in private, I set out to try to nudge the insider culture back into a more inclusive, community focused stewardship. Unfortunately, personal circumstances made my progress slow and my timing became terrible.

When the hyper-growth that started about a year ago came with enormous expectations and attention by the public and the press, looking good became more attractive than actually doing good. With that comes a whole shipload of problems because it makes developers conservative towards their status and legacy at the cost of the thing they were meant to maintain: the cryptocurrency itself. When only the appearance counts, more time is spent on creating narrative and forming stronger echo-chambers of confirmation, than actually making sure that the coin survives.

I feel that because the development of Dogecoin Core has evolved away from the path of being open and driven by collaborative work, and more into a thing of individual power and fame, the continuity of the whole crypto is slowly exposed to increased risk. Eggs are being concentrated in very few baskets and this is the biggest risk to Dogecoin today: a broad and open Dogecoin Core will guarantee its continuity, but an exclusive group of devs will burn out (or run out of money) and risk obsolescence for the entire ecosystem.

Forget about 51% attacks, government censoring, regulation and coin tainting; none of that matters if there is no good reference software that defines consensus rules. Over the past year it was a bit of a struggle to get Dogecoin Core back into shape after the version 4 soft fork forced everyone to use 1.14.2 in November 2020. I hope that every shibe can appreciate not losing another year.

### The current situation

Last week, Ross announced [his stepping back from Dogecoin development](https://rnicoll.name/posts/2022-02-16-stepping-away/). This is disruptive because over the last year, Ross has reviewed a record 163 pull requests on Dogecoin Core, whereas I reviewed 144. Combined, he and I have done approximately 95% of all review work, making his leaving a challenge that we’ll have to overcome. However, this not the end of Dogecoin Core, on the contrary: it's an opportunity to do better, be more inclusive, and depend less on a central team. There’s nothing wrong with centralized channels because we will always need those for people to collaborate. However, these channels must be open and transparent, much more than they have been the past couple of years.

The current approval process for pull requests, that has been in place since 2014, is that under a 4-eye principle, no developer can review and approve their own work, and that every pull request **must** be reviewed and approved by a maintainer. I think those rules are pretty good but the execution will now become problematic. However, this does not have to be a big problem: we simply have to get more eyes on proposals.

### Proposed way forward

I have polled a couple of contributors about their willingness to do reviews and so far the unanimous reaction has been that people are absolutely willing to help out, but need guidance. In the tradition of Dogecoin, the best thing to do when you face a difficult problem is to steal an idea from Bitcoin, in this case the most suiting vehicle to steal from is the [Bitcoin Core PR Review Club](https://bitcoincore.reviews/). I propose to do 2 PR review sessions to see if this helps, starting the week of February 27th: the first one for [#2698](https://github.com/dogecoin/dogecoin/pull/2698) and perhaps [#2773](https://github.com/dogecoin/dogecoin/pull/2773) as a second one. I’m considering to do the meetings on IRC rather than Discord, because it is less intrusive and you don’t need a client for that (i.e. web clients work well) and it's only just for an hour per week at most. Please let us all know if you think there is a compelling reason not to use IRC.

At the same time, we have an immediate need for more active maintainers because we need to get work merged. To that end I have, about an hour ago, proposed to the other maintainers that we grant maintainer rights to Chromatic as they have been the third most active developer for Dogecoin Core the past year, are experienced in C++ development, and have contributed not just towards the easy or make-them-look-good work, but also to the annoying things that just needed to get done. These are good characteristics for a maintainer to have, because the not-sexy work often falls to them. What this means is that under the above-mentioned 4-eye principle, we gain 2 much-needed active eyes, and I hope that through the PR reviews, we can attract more enthusiast reviewers.

🚀

**Comment by qlpqlp** [comment]
Great to see that, "what doesn't kill you makes you stronger " reading :)

Hope to see more and more people jumping in this adventure 😉

Dogecoin is here to stay, and it just needs one Shibe to believe in  it, if more belives in it, even better 😅

To bad my C++ skils where not improved/maintained over the years wen i started in C, then C++, then VB, finally PHP until today, to help in outher forms.

**Comment by hadar33** [comment]
My c++ skills are rusty, however i am willing todo anything to help.

P.S. IRC is a great idea

**Comment by shibe2** [comment]
@patricklodder Your link https://web.archive.org/web/20210209140212/https://www.reddit.com/r/dogecoin/comments/22hqm6/official_an_update_on_the_foundation_core does not have any information about Dogecoin Core copyright. I looked up the [original post](https://web.archive.org/web/20140411193346/http://www.reddit.com/r/dogecoin/comments/22hqm6/official_an_update_on_the_foundation_core) that is now deleted, and it doesn't discuss copyright either.

**Reply by patricklodder** [reply]
The copyright of the code, as a product of the "project", has been copyrighted under the MIT license to "Dogecoin developers" (eg [1.4.1](https://github.com/dogecoin/dogecoin/blob/v1.4.1/COPYING), [1.6.1](https://github.com/dogecoin/dogecoin/blob/v1.6.1/COPYING) and [1.8.3](https://github.com/dogecoin/dogecoin/blob/v1.8.3/COPYING), with some branches like 1.10 and the initial 1.14 having omitted updates to the main license file, which is kind of sloppy)

Perhaps my usage of the term "public domain" is wrong here, and I should use another term to describe the end of the tenure of the project team and the beginning of a new era where "you are Dogecoin" was the statement given to the community. I don't know what term better describes that though. How would you describe that?

**Reply by patricklodder** [reply]
PS: I linked the wrong snapshot - the one that said deleted instead of the original - so I fixed that.

**Reply by shibe2** [reply]
Transfer control of the project to larger developer community.

**Reply by patricklodder** [reply]
> Transfer control of the project to larger developer community.

Part of the project (the development side) yes. I'll do an attempt to make it more concise.

**Reply by Dorminus** [reply]
Hi everybody, 

I know this is an old thread but I’m not sure where to go, or who to contact. I stumbled across a recent “Dogecoin” twitter post that asked for help with translations. I can help with german if needed. 

Kind regards

**Reply by patricklodder** [reply]
@Dorminus that would be awesome. Here's how you do it:

1. [fork this repository](https://github.com/dogecoin/dogecoin/fork)
2. In your fork, browse to the 1.14.7-dev branch, select it and then create a new branch for your updates, like so: <img width="304" alt="Screen Shot 2022-08-19 at 6 07 06 PM" src="https://user-images.githubusercontent.com/1410115/185660920-f1cdf2ee-c3a0-4c70-86c9-504c76c998bc.png">
3. Edit the file `src/qt/locale/bitcoin_de.ts`
4. When you're done, open a pull request from your newly created branch into 1.14.7-dev on this repository (it should show you a green "compare and pull request" button

**Comment by chromatic** [comment]
:+1: on a review club, especially for substantial PRs like the ones linked.

+0 on IRC; whatever lets people coordinate and participate is useful.

**Comment by michilumin** [comment]
So I think it's clear we do need more input and 'processing surface' of PRs and issues that need to be addressed.  Discord is targeted by spammers and "pump bois" way too often so I can see the purpose for IRC as a bit of a 'serious business' filter.  The only downside I can see is "more areas to keep track of".  (Everyone has their own favorite platform, which is fair, but it just ends up that there are a lot of them.) 

I see no harm at all in a review club; The more eyes the better on reviews; though I suppose it could lead to a lower signal to noise ratio too. (i.e. protracted exceptions on how things should be implemented. Though there's a lot of noise now; i think at this point that reward outweighs the risk.)

I'm trying to get back on the horse on participation in github. Which is why I'm making a point to reply here. (Installed the app on all my phones JUST for the purpose of github notifications lately, which has been unideal previously, hopefully this should help.) 

If Chromatic has done that legwork of living/eating/breathing the repo, then yes, especially with the loss of Ross, i could see good rationale for him being a maintaner.

**Reply by patricklodder** [reply]
Thanks!

### Re: signal to noise. 

Leaving a review is always good if there is a reason stated, and per the [contribution guidelines](https://github.com/dogecoin/dogecoin/blob/master/CONTRIBUTING.md#peer-review), reviews are already open to anyone that wants to leave one. (And in case you fear a total flood of meaningless reviews, it also explicitly states that maintainers can weigh reviews based on merit and common sense in case there are disagreements.) I think it's in general better to be aware of any objections before merge and be able to consider them, than to merge quickly and then get a comment on the commit, after the fact - or worse, get another PR because review missed something.

We've seen some shibes the past year approve pull requests for mysterious reasons (just add an approval, but no rationale or details given) and this is noisy, especially when they happen on pull requests that fail CI or otherwise aren't complete yet. I think that in both cases, either if these are meant to be signal and not noise, we'll actually increase the signal and reduce the noise, because people simply don't understand how it works, or, if they are in fact noise, then getting more meaningful reviews will also increase the signal to noise ratio.

Bottom line, I think that this could cause an improvement in the quality of the reviews too.

**Comment by Much-Mining** [comment]
> ### Ancient history



> 
> ### Proposed way forward
> In the tradition of Dogecoin, the best thing to do when you face a difficult problem is to steal an idea from Bitcoin, in this case the most suiting vehicle to steal from is the [Bitcoin Core PR Review Club](https://bitcoincore.reviews/). 

Lol. Yayyyy!

>I’m considering to do the meetings on IRC rather than Discord

I don't know about coding or what I could do to assist this but having an IRC would be a cool from my perspective. It gives me a nice sccreenshot to nerd out to on the Twitter. & IRC Keeps it simple. 

> 🚀

🚀
